lib/ohai/plugins/linux/filesystem.rb

```diff
@@ -142,36 +143,48 @@ def generate_mountpoint_view(fs)
       end
     end
 
-    have_lsblk = File.exist?("/bin/lsblk")
+    # We used to try to decide if we wanted to run lsblk or blkid
+    # but they each have a variety of cases were they fail to report
+    # data. For example, there are a variety of cases where lsblk won't
+    # report unmounted filesystems, but blkid will. And vise-versa. Sweet.
+    # So for reliability, we'll run both, if we have them.
+
+    have_lsblk = File.exist?("/bin/lsblk") || File.exist?("/usr/bin/lsblk")
+    have_blkid = File.exist?("/bin/blkid") || File.exist?("/usr/bin/blkid")
+    cmds = []
+    # These should be in order of preference... first writer wins.
     if have_lsblk
-      cmd = "lsblk -n -P -o NAME,UUID,LABEL,FSTYPE"
-    else
-      # CentOS5 and other platforms don't have lsblk
-      cmd = "blkid"
+      cmds << "lsblk -n -P -o NAME,UUID,LABEL,FSTYPE"
+    end
+    if have_blkid
+      cmds << "blkid"
     end
 
-    so = shell_out(cmd)
-    so.stdout.each_line do |line|
-      parsed = parse_line(line, have_lsblk)
-      next if parsed.nil?
-      # lsblk lists each device once, so we need to update all entries
-      # in the hash that are related to this device
-      keys_to_update = []
-      fs.each_key do |key|
-        keys_to_update << key if key.start_with?("#{parsed[:dev]},")
-      end
+    cmds.each do |cmd|
+      cmdtype = File.basename(cmd.split.first)
```

Comment:
Um. Sure? I kinda prefer having it tied to the actual command rather than people having to keep things in sync by typing twice.